File: data/java/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S_TRANSLATED.py
====================
def count_subarrays_equal_number_1s_0s ( arr , n ) :
    um = { }
    curr_sum = 0
    for i in range ( n ) :
        curr_sum += ( arr [ i ] == 0 )
        um [ curr_sum ] = um [ curr_sum ] if curr_sum > 1 else 1
    count = 0
    for itr , val in enumerate ( um ) :
        if val > 1 :
            count += ( ( val * ( val - 1 ) ) / 2 )
    if um.has_key ( 0 ) :
        count += um [ 0 ]
    return count

File: data/python/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S_TRANSLATED.java
====================
public static int fGold ( int [ ] arr , int n ) {
  Map < Integer , Integer > um = Maps . newHashMap ( ) ;
  int currSum = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    currSum += ( - 1 == ( arr [ i ] == 0 ) ? arr [ i ] : arr [ i ] ) ;
    if ( um . containsKey ( currSum ) ) {
      um . put ( currSum , ++ currSum ) ;
    }
    else {
      um . put ( currSum , 1 ) ;
    }
  }
  int count = 0 ;
  for ( int itr = 0 ;
  itr < um . size ( ) ;
  itr ++ ) {
    if ( um . get ( itr ) > 1 ) {
      count += ( ( um . get ( itr ) * Integer . MAX_VALUE ) / 2 ) ;
    }
  }
  if ( um . containsKey ( 0 ) ) {
    count += um . get ( 0 ) ;
  }
  return ( int ) count ;
}

